velox/expression/SignatureBinder.cpp

```diff
@@ -111,63 +118,134 @@ bool checkNamedRowField(
   return true;
 }
 
+bool checkSignatureProperties(
+    const SignatureVariable& signature,
+    const TypePtr& actualType) {
+  if (signature.knownTypesOnly() && actualType->isUnKnown()) {
+    return false;
+  }
+  if (signature.orderableTypesOnly() && !actualType->isOrderable()) {
+    return false;
+  }
+  if (signature.comparableTypesOnly() && !actualType->isComparable()) {
+    return false;
+  }
+  return true;
+}
+
 } // namespace
 
-bool SignatureBinder::tryBindWithCoercions(std::vector<Coercion>& coercions) {
-  return tryBind(true, coercions);
-}
+// Traverse all types in signatures and deduce the least common type for each
```

Comment:
this comment belongs to .h file; please, move

Reply:
Done, see https://github.com/facebookincubator/velox/pull/15734/changes/861117750d189ac36413499955e1c28d07e1fb42